--- origsrc/solfege-3.10.4/soundcard/Makefile	2008-04-29 12:31:56.000000000 -0500
+++ src/solfege-3.10.4/soundcard/Makefile	2008-05-14 22:43:56.218750000 -0500
@@ -24,18 +24,18 @@
 
 easy-dist-files += soundcard/solfege_c_midi_wrap.c
 
-CFLAGS:=$(CFLAGS) -I/usr/src/linux/include -Wall -pedantic -fPIC
+CFLAGS:=$(CFLAGS) -Wall -pedantic
 CC:=$(CC) $(CFLAGS) -DHAVE_CONFIG_H 
 
 ifeq ($(ENABLE_OSS_SOUND),yes)
  ifeq ($(SWIG_VERSION),1.1)
-  TARGETS += soundcard/solfege_c_midimodule.so
+  TARGETS += soundcard/solfege_c_midimodule.dll
  endif
  ifeq ($(SWIG_VERSION),1.3)
-  TARGETS += soundcard/_solfege_c_midi.so
+  TARGETS += soundcard/_solfege_c_midi.dll
  endif
  ifeq ($(SWIG_VERSION),unknown)
-  TARGETS += soundcard/_solfege_c_midi.so
+  TARGETS += soundcard/_solfege_c_midi.dll
  endif
 endif
 
@@ -47,11 +47,11 @@
 OBJECTS += fft.o dsp.o gate.o calc.o xmalloc.o
 endif
 
-soundcard/solfege_c_midimodule.so: $(OBJECTS)
-	$(CC) -fPIC -shared $(OBJECTS) -o $@ -lc
+soundcard/solfege_c_midimodule.dll: $(OBJECTS)
+	$(CC) -shared $(OBJECTS) -o $@ -lpython$(PYTHON_VERSION)
 
-soundcard/_solfege_c_midi.so: $(OBJECTS)
-	$(CC) -fPIC -shared $(OBJECTS) -o $@
+soundcard/_solfege_c_midi.dll: $(OBJECTS)
+	$(CC) -shared $(OBJECTS) -o $@ -lpython$(PYTHON_VERSION)
 
 soundcard/solfege_c_midi_wrap.o: soundcard/solfege_c_midi_wrap.c soundcard/solfege_c_midi.i
 	(cd soundcard; $(CC) $(CFLAGS) -c solfege_c_midi_wrap.c $(PYTHON_INCLUDES))
@@ -80,21 +80,21 @@
 ifeq ($(ENABLE_OSS_SOUND),yes)
 	mkdir -p $(DESTDIR)/$(libdir)/$(PACKAGE)
 ifeq ($(SWIG_VERSION),1.1)
-	$(INSTALL_DATA) soundcard/solfege_c_midimodule.so $(DESTDIR)/$(libdir)/$(PACKAGE)/
+	$(INSTALL_DATA) soundcard/solfege_c_midimodule.dll $(DESTDIR)/$(libdir)/$(PACKAGE)/
 else
-	$(INSTALL_DATA) soundcard/_solfege_c_midi.so $(DESTDIR)/$(libdir)/$(PACKAGE)/
+	$(INSTALL_DATA) soundcard/_solfege_c_midi.dll $(DESTDIR)/$(libdir)/$(PACKAGE)/
 endif
 endif
 
 uninstall-soundcard:
 ifeq ($(SWIG_VERSION),1.1)
-	rm -f $(DESTDIR)/$(libdir)/$(PACKAGE)/solfege_c_midimodule.so
+	rm -f $(DESTDIR)/$(libdir)/$(PACKAGE)/solfege_c_midimodule.dll
 else
-	rm -f $(DESTDIR)/$(libdir)/$(PACKAGE)/_solfege_c_midi.so
+	rm -f $(DESTDIR)/$(libdir)/$(PACKAGE)/_solfege_c_midi.dll
 endif
 	rmdir $(DESTDIR)/$(libdir)/$(PACKAGE) || true
 	rmdir $(DESTDIR)/$(libdir)/$(PACKAGE) || true
 
 clean-files += \
-	$(wildcard soundcard/*.o soundcard/*.so soundcard/*.pyc soundcard/*.pyd soundcard/*.pyo)
+	$(wildcard soundcard/*.o soundcard/*.dll soundcard/*.pyc soundcard/*.pyd soundcard/*.pyo)
 
